src/layer/marker/Icon.js

```diff
@@ -47,8 +47,14 @@ L.Icon = L.Class.extend({
 	},
 
 	_setIconStyles: function (img, name) {
-		var options = this.options,
-		    size = L.point(options[name + 'Size']),
+		var options = this.options;
+		var sizeOption = options[name + 'Size'];
+
+		if (!L.Util.isArray(sizeOption)) {
+			options[name + 'Size'] = [options[name + 'Size'], options[name + 'Size']];

```

Comment:
Also, you're overwriting the original options object, which is undesirable. Instead you could redefine `sizeOption`.